analyzers/src/SonarAnalyzer.CSharp/Rules/TestClassShouldHaveTestMethod.cs

```diff
@@ -65,19 +65,34 @@ public sealed class TestClassShouldHaveTestMethod : SonarDiagnosticAnalyzer
                 SyntaxKind.ClassDeclaration,
                 SyntaxKindEx.RecordClassDeclaration);
 
-        private static bool HasAnyTestMethod(INamedTypeSymbol classSymbol) =>
-            classSymbol.GetMembers().OfType<IMethodSymbol>().Any(m => m.IsTestMethod());
+        private static bool HasAnyTestMethod(INamespaceOrTypeSymbol symbol) =>
+            symbol.GetMembers().OfType<IMethodSymbol>().Any(m => m.IsTestMethod());
 
-        private static bool IsViolatingRule(INamedTypeSymbol classSymbol) =>
-            classSymbol.IsTestClass()
-            && !HasAnyTestMethod(classSymbol);
+        private static bool IsViolatingRule(INamedTypeSymbol symbol) =>
+            symbol.IsTestClass()
+            && !HasAnyTestMethod(symbol);
 
-        private static bool IsExceptionToTheRule(INamedTypeSymbol classSymbol) =>
-            classSymbol.IsAbstract
-            || (classSymbol.BaseType.IsAbstract && HasAnyTestMethod(classSymbol.BaseType))
-            || HasSetupOrCleanupAttributes(classSymbol);
+        private static bool IsExceptionToTheRule(ITypeSymbol symbol) =>
+            symbol.IsAbstract
+            || HasBaseTypeWithTestMethods(symbol)
```

Comment:
We have `symbol.GetSelfAndBaseTypes()` that we can use. So this could look something like
```
|| symbol.GetSelfAndBaseTypes().Any(HasAnyTestMethod)
```